### PR TITLE
Add benchmarks for opam show and String.split

### PR DESCRIPTION
**Pull request by kit-ty-kate**
This PR is meant to show the performance gain from https://github.com/ocaml/opam/pull/6210

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
https://github.com/ocaml/opam/pull/6210 is unable to show any gain. I'll convert this PR to a draft while this is being examined.

File: tests/bench/bench.ml [hunk 1]
```diff
@@ -99,6 +99,39 @@ let () =
...
+      let ic = Stdlib.open_in_bin "/home/opam/all-opam-content" in
+      let rec loop files =
+        match Stdlib.input_line ic with
+        | file -> loop (file :: files)
+        | exception End_of_file -> files
+      in
+      loop []
+    in
+    let before = Unix.gettimeofday () in
+    List.iter (fun line -> ignore (OpamStd.String.split line ' ')) lines;
```
Comment (rjbou) [inline tests/bench/bench.ml:132]: Wouldn't be more precise if it is run 50, 100 times ? Current run/time is less than 1 second.

Comment (kit-ty-kate) [reply tests/bench/bench.ml:132]: the problem is that it is only 1s in #6210. In this PR it takes 45s which would be too long to run 50 or 100 times.

Comment (kit-ty-kate) [reply tests/bench/bench.ml:132]: i can run it 10 times maybe? That would be 4.5mins for this benchmark which seems still reasonable

Comment (rjbou) [reply tests/bench/bench.ml:132]: yes, the idea was to not have to diff times under 1s, to have more "accurate" data.

Comment (rjbou) [reply tests/bench/bench.ml:132]: The 5min bench is just for few launches

Comment (kit-ty-kate) [reply tests/bench/bench.ml]: done

File: tests/bench/bench.ml [hunk 2]
```diff
@@ -148,6 +185,31 @@ let () =
...
+          "value": %f,
+          "units": "secs"
+        },
+        {
+          "name": "opam show --raw pkgname.version",
+          "value": %f,
+          "units": "secs"
+        },
+        {
+          "name": "OpamStd.String.split",
```
Comment (rjbou) [inline tests/bench/bench.ml:210]:
```suggestion
          "name": "OpamStd.String.split amortised over 10 runs",
```

**Review by rjbou — APPROVED** [review]

**Pull request merged**